### ISSUE TITLE
Load order

### ISSUE THREAD

**Issue by marcandre** [guillaumepotier/Parsley.js#976]
Can we deprecate doing anything before loading `Parsley`?

I don't really see an upside, there shouldn't be any need for it. Load `parsley`, then do whatever you want with it.

Ideally, there would be an actual API for stuff. This helps for compatibility going forward, allows us to catch common errors, etc.

**Comment by marcandre** [comment]
I'll add to my list of things I'd like to deprecate: `ParsleyExtend`. I'd give access to the actual classes as follows: `window.Parsley.Field`, etc. and state that changing the prototype is a possibility but no compatibility intended. Every normal use of `ParsleyExtend` should have an API (e.g. `window.Parsley.addMessage(...)`)

I'd also deprecate `ParsleyConfig` and use `Parsley.options` instead.

If there are any other remaining global constants (other that `Parsley`), we should look into removing them. What do you think, @guillaumepotier ?

**Comment by guillaumepotier** [comment]
Agreed with you. Thought it could be handy, but turns to be harder to maintain (sometimes, 2 different ways to do the same thing).

But maybe we should wait es6 branch before doing so right?

**Comment by marcandre** [comment]
:+1:

**Comment by porjo** [comment]
As a new Parsley user, I found the current load order behaviour confusing. For example, I was trying to do this:

```JS
$('#form').parsley();
$('.fields').attr({'data-parsley-maxlength': 4, 'data-parsley-trigger': 'change'});
```
...and wondering why it wasn't working. IMHO, the documentation doesn't make it clear that loading Parsley should come afterwards.

**Comment by marcandre** [comment]
@porjo That's not really the load order, but is the initialization order. PRs on documentation always welcome
